File: mobile/lib/models/product_image.dart
class ProductImage {
  final int id;
  final int productId;
  final String originalUrl;
  final String? processedUrl;
  final DateTime? createdAt;

  const ProductImage({
    required this.id,
    required this.productId,
    required this.originalUrl,
    this.processedUrl,
    this.createdAt,
  });

  factory ProductImage.fromJson(Map<String, dynamic> json) {
    return ProductImage(
      id: json['id'] as int? ?? 0,
      productId: json['product_id'] as int? ?? 0,
      originalUrl: json['original_url'] as String? ?? '',
      processedUrl: json['processed_url'] as String?,
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'].toString())
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'product_id': productId,
      'original_url': originalUrl,
      'processed_url': processedUrl,
      'created_at': createdAt?.toIso8601String(),
    };
  }
}

File: backend/app/models/product_image.py
from datetime import datetime
from typing import TYPE_CHECKING, Optional
from sqlalchemy import String, Integer, ForeignKey, DateTime, func
from sqlalchemy.orm import Mapped, mapped_column, relationship
from backend.app.db.base import Base

if TYPE_CHECKING:
    from backend.app.models.product import Product

class ProductImage(Base):
    __tablename__ = "product_images"

    id: Mapped[int] = mapped_column(primary_key=True, index=True, autoincrement=True)
    product_id: Mapped[int] = mapped_column(
        Integer,
        ForeignKey("products.id", ondelete="CASCADE"),
        index=True,
        nullable=False
    )
    original_url: Mapped[str] = mapped_column(String(1024), nullable=False)
    processed_url: Mapped[Optional[str]] = mapped_column(String(1024), nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False
    )

    # Relationships
    product: Mapped["Product"] = relationship("Product", back_populates="images")

    def __repr__(self) -> str:
        return f"<ProductImage(id={self.id}, product_id={self.product_id}, original_url='{self.original_url}')>"
